psl/psl-core/src/datamodel_connector.rs

```diff
@@ -284,6 +296,25 @@ pub trait Connector: Send + Sync {
 
     fn push_completions(&self, _db: &ParserDatabase, _position: SchemaPosition<'_>, _completions: &mut CompletionList) {
     }
+
+    fn parse_datasource_properties(
+        &self,
+        args: &mut HashMap<&str, (Span, &ast::Expression)>,
+        diagnostics: &mut Diagnostics,
+    ) -> DatasourceConnectorData {
+        if let Some((span, _)) = args.remove(EXTENSIONS_KEY) {
+            diagnostics.push_error(DatamodelError::new_static(
+                "The `extensions` property is only available with the `postgresql` connector.",
+                span,
+            ));
+        }
+
+        Default::default()
+    }
+
+    fn render_datasource_properties(&self, _: &DatasourceConnectorData, _: &mut String) -> fmt::Result {
```

Comment:
I think we should avoid relying on `psl::Datasource` for rendering in introspection. Let's discuss that this afternoon.

Reply:
Do we want to refactor it in this PR or later?

Reply:
We can do it later, depending on what we decide for the datasource extensions